docs/design/mscclpp-dsl-json-file.md

```diff
@@ -0,0 +1,478 @@
+- ```type```: Specifies the type of the channel, which in the case of the Memory Channel, will be ```port```.
+- ```connectedTo```: Specifies the connections between channels. For example, if we have a list like: [1, 2, 3], it indicates that there are three channels: the first is connected to rank 1, the second to rank 2, and the third to rank 3.
+
+Example:
+
+```json
+"channels": [
+  {
+    "type": "port",
+    "connectedTo": [
+      1,
+      2,
+      3
+    ]
+  }
+]
+```
+
+The Switch Channel has the following fields:
+- ```type```: Specifies the type of the channel, which in the case of the Memory Channel, will be ```switch```.
+- ```buff```: Consist of the buffer type which the Switch Channel will be binded, this could have the following values: "i" for the input buffer, "o" for the output buffer, "s" for the scratch buffer.
+- ```rankGroups```: Consist of the group of the ranks connected by the Switch Channel, this field contains the size and the list of the ranks connected.
+
+Example:
+
+```json
+"channels": [
+  {
+    "type": "switch",
+    "buff": "i",
+    "rankGroups": [
+      {
+        "size": 8,
+        "ranks": [
+          0,
+          1,
+          2,
+          3,
+          4,
+          5,
+          6,
+          7
+        ]
+      }
+    ]
+  }
+]
+```
+
+### Buffer Collection
+The buffer collection field will contain the description of all the remote buffers that his GPU need to access, it will contain the following fields:
+
+-
+-
+-
+
+Example
+
+```json
+"bufferCollection": [
+  {
+    "rank": 1,
+    "buff": "i"
+  }
+]
+```
+
+### Thread Block
+The thread block field describe the operation inside each thread block, we have the following fields:
+
+- ```id```: The thread block id.
+- ```ops```: The list with all the operation in order they will be executed by this tread block.
+- ```channels```: The channels the thread block will use, referenced by the channel id. The channel id is based on the global channel description in the gpu, for example if the channel type is memory and the channel id is 0, it refers to the first channel of memory id type descriptioned in the gpu channels field.
+- ```bufferCollection```: A list with all the remote buffers used by the thread block.
+
+For Example:
+
+```json
+"threadblocks": [
+  {
+    "id": 0,
+    "ops": [...],
+    "channels":[
+      {
+        "ctype": "memory",
+        "cids": [
+          0
+        ]
+      }
+    ],
+    "bufferCollection": [
+      {
+        "buffid": 0
+      }
+    ]
+  }
+]
+```
+
+The most important field here is the ops fields, where there is the description for all the operations, each operation has specific fields, let's given an overview in all the possible fields and after this go throught each of them:
+
+- ```name```: Operation name, it include: signal, wait, put, get.
+- ```ctype```: Channel type, it include: memory, port, switch.
+- ```i_cids```: The channel id of the channels used in the receive operations.
```

Comment:
Since we decouple channel and buffer, maybe we can just use cids for both i_cids and o_cids